<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="malbacR">
<title>malbacR: A Vignette Pertaining to Pre-Filtered Data • malbacR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="malbacR: A Vignette Pertaining to Pre-Filtered Data">
<meta property="og:description" content="malbacR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">malbacR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/malbacR_vignette_condensed_filtered.html">malbacR: A Vignette Pertaining to Pre-Filtered Data</a>
    <a class="dropdown-item" href="../articles/malbacR_vignette_expanded_unfiltered.html">malbacR: A Vignette Pertaining to Unfiltered Data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>malbacR: A Vignette Pertaining to Pre-Filtered Data</h1>
                        <h4 data-toc-skip class="author">Damon Leach,
Kelly Stratton, Lisa Bramer</h4>
            
            <h4 data-toc-skip class="date">2023-10-18</h4>
      
      
      <div class="d-none name"><code>malbacR_vignette_condensed_filtered.Rmd</code></div>
    </div>

    
    
<p>The R package <code>malbacR</code> is a new package that deals with
batch correction methods of small molecule omics data. It works in
conjunction with <code>pmartR</code>, an R package designed for
preprocessing, filtering, and analyzing multi-omics data <span class="citation">(Stratton et al. 2019; Degnan et al. 2023)</span>.</p>
<p>This outline demonstrates the functionality and use of the
<code>malbacR</code> package. Within this package, there are four data
sets: <code>pmart_amide</code>,
<code>pmart_amideFilt</code>,<code>pmart_mix</code>, and
<code>pmart_mixFilt</code>. The first two objects contain data
originally found in the package <code>WaveICA2.0</code> <span class="citation">(Deng and Li 2022)</span>. The second two objects
contain data originally found in the package <code>crmn</code> <span class="citation">(Redestig et al. 2009)</span>. The data sets with the
suffix “Filt” are filtered versions of the original data sets. That is,
unlike their non-filtered counterparts, these data objects require no
additional data manipulation for batch correction to successfully run.
For this example, we work with the filtered data sets to demonstrate how
the <code>malbacR</code> functions work.</p>
<p>Within <code>malbacR</code> there are 11 batch correction
methods:</p>
<ol style="list-style-type: decimal">
<li>Range Scaling <span class="citation">(Berg et al. 2006)</span>
</li>
<li>Power Scaling <span class="citation">(Berg et al. 2006)</span>
</li>
<li>Pareto Scaling <span class="citation">(Berg et al. 2006)</span>
</li>
<li>ComBat <span class="citation">(Müller et al. 2016)</span>
</li>
<li>EigenMS <span class="citation">(Karpievitch et al. 2014)</span>
</li>
<li>NOMIS <span class="citation">(Sysi-Aho et al. 2007)</span>
</li>
<li>RUV-random <span class="citation">(De Livera et al.
2015)</span>
</li>
<li>QC-RLSC <span class="citation">(Dunn et al. 2011)</span>
</li>
<li>WaveICA2.0 <span class="citation">(Deng et al. 2019)</span>
</li>
<li>TIGER <span class="citation">(Han et al. 2022)</span>
</li>
<li>SERRF <span class="citation">(Fan et al. 2019)</span>
</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">malbacR</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pmartR/pmartR" class="external-link">pmartR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="data-set-1-amide-data">Data Set 1: Amide Data<a class="anchor" aria-label="anchor" href="#data-set-1-amide-data"></a>
</h2>
<div class="section level3">
<h3 id="load-in-the-data">Load in the Data<a class="anchor" aria-label="anchor" href="#load-in-the-data"></a>
</h3>
<p>A <code>pmartR</code> friendly version of the “Amide” data that has
already undergone filtering and log2 transformations is already
implemented within <code>malbacR</code>. Therefore, we simply load in
the data.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pmart_amideFilt</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-batch-correction-methods">Run Batch Correction Methods<a class="anchor" aria-label="anchor" href="#run-batch-correction-methods"></a>
</h3>
<p>The “Amide” data set contains information regarding quality control
samples. Therefore the following methods can be used:</p>
<ul>
<li>Range Scaling</li>
<li>Power Scaling</li>
<li>Pareto Scaling</li>
<li>TIGER</li>
<li>EigenMS</li>
<li>WaveICA2.0</li>
<li>SERRF</li>
<li>ComBat</li>
</ul>
<p>It is important to note that the data <code>pmart_amideFilt</code>
has been filtered, transformed, and imputed such that the data runs with
all possible methods, but it may be conservative for a given method. For
example, some methods do not require imputation to run, but because some
methods require imputation (like SERRF and WaveICA2.0) <span class="citation">(Fan et al. 2019; Deng et al. 2019)</span>, the data
has been imputed.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># SCALING METHODS</span></span>
<span><span class="co"># range scaling</span></span>
<span><span class="va">amide_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_range.html">bc_range</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># power scaling</span></span>
<span><span class="va">amide_power</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_power.html">bc_power</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># pareto scaling</span></span>
<span><span class="va">amide_pareto</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_pareto.html">bc_pareto</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># QUALITY CONTROL METHODS</span></span>
<span><span class="co"># TIGER</span></span>
<span><span class="va">amide_tiger</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_tiger.html">bc_tiger</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span>,sampletype_cname <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                        test_val <span class="op">=</span> <span class="st">"QC"</span>,injection_cname <span class="op">=</span> <span class="st">"Injection_order"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># QC-RLSC</span></span>
<span><span class="va">amide_qcrlsc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_qcrlsc.html">bc_qcrlsc</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span>,block_cname <span class="op">=</span> <span class="st">"batch"</span>,</span>
<span>                          qc_cname <span class="op">=</span> <span class="st">"group"</span>, qc_val <span class="op">=</span> <span class="st">"QC"</span>, order_cname <span class="op">=</span> <span class="st">"Injection_order"</span>,</span>
<span>                          missing_thresh <span class="op">=</span> <span class="fl">0.5</span>, rsd_thresh <span class="op">=</span> <span class="fl">0.3</span>, backtransform  <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># SERRF</span></span>
<span><span class="va">amide_serrf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_serrf.html">bc_serrf</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span>,sampletype_cname <span class="op">=</span> <span class="st">"group"</span>,test_val <span class="op">=</span> <span class="st">"QC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># OTHER METHODS</span></span>
<span><span class="co"># ComBat</span></span>
<span><span class="va">amide_combat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_combat.html">bc_combat</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span>, use_groups <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># EigenMS</span></span>
<span><span class="va">amide_eigen</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_eigenMS.html">bc_eigenMS</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># WaveICA2.0</span></span>
<span><span class="va">amide_wave</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_waveica.html">bc_waveica</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amideFilt</span>, injection_cname <span class="op">=</span> <span class="st">"Injection_order"</span>,</span>
<span>                         alpha <span class="op">=</span> <span class="fl">0</span>, cutoff <span class="op">=</span> <span class="fl">0.1</span>, K <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-visualization">Data Visualization<a class="anchor" aria-label="anchor" href="#data-visualization"></a>
</h3>
<p>After obtaining all the different batch corrected data sets, we can
plot the PPCA to see if they are successfully returning batch corrected
data. We set the <code>group_designation</code> to use our batch
information as the <code>main_effects</code> so as to color the data by
batch. All of this code is run using functions from <code>pmartR</code>
demonstrating the utility between the two packages.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pmart_amide</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">pmart_amideFilt</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_range</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_power</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_power</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_pareto</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_pareto</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_tiger</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_tiger</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_qcrlsc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_qcrlsc</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_combat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_combat</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_wave</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_wave</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span><span class="va">amide_serrf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">amide_serrf</span>,main_effects <span class="op">=</span> <span class="st">"batch"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_amide</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: Unadjusted"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_range</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: Range"</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_power</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: Power"</span><span class="op">)</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_pareto</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: Pareto"</span><span class="op">)</span></span>
<span><span class="va">p5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_tiger</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: TIGER"</span><span class="op">)</span></span>
<span><span class="va">p6</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_qcrlsc</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: QC-RLSC"</span><span class="op">)</span></span>
<span><span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_combat</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: ComBat"</span><span class="op">)</span></span>
<span><span class="va">p8</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_wave</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: WaveICA2.0"</span><span class="op">)</span></span>
<span><span class="va">p9</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">amide_serrf</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Amide: SERRF"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span>;<span class="va">p2</span>;<span class="va">p3</span>;<span class="va">p4</span>;<span class="va">p5</span>;<span class="va">p6</span>;<span class="va">p7</span>;<span class="va">p8</span>;<span class="va">p9</span></span></code></pre></div>
<p><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-1.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-2.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-3.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-4.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-5.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-6.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-7.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-8.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-5-9.png" width="33%"></p>
</div>
</div>
<div class="section level2">
<h2 id="data-set-2-mix-data">Data Set 2: Mix Data<a class="anchor" aria-label="anchor" href="#data-set-2-mix-data"></a>
</h2>
<div class="section level3">
<h3 id="load-in-the-data-1">Load in the Data<a class="anchor" aria-label="anchor" href="#load-in-the-data-1"></a>
</h3>
<p>The <code>malbacR</code> package includes another <code>pmartR</code>
friendly data set for the “mix” data which has already undergone
filtering and log2 transformations is already implemented within
<code>malbacR</code>. Therefore, we simply load in the data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pmart_mixFilt</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-batch-correction-methods-1">Run Batch Correction Methods<a class="anchor" aria-label="anchor" href="#run-batch-correction-methods-1"></a>
</h3>
<p>The “mix” data set contains information regarding negative
controls/internal standards. Therefore the following methods can be
used:</p>
<ul>
<li>Range Scaling</li>
<li>Power Scaling</li>
<li>Pareto Scaling</li>
<li>RUV-random</li>
<li>NOMIS</li>
<li>ComBat</li>
</ul>
<p>As there is no information regarding QC samples or injection order,
the other batch correction methods in the package or not able to be used
with this data set.</p>
<p>As with the <code>pmart_amideFilt</code> data set, is important to
note that the data in <code>pmart_mixFilt</code> has been filtered and
transformed such that the data runs with all possible methods, but it
may be conservative for a given method. As there were no missing data
with the “mix” data set, no imputation was computed.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># SCALING METHODS</span></span>
<span><span class="co"># range scaling</span></span>
<span><span class="va">mix_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_range.html">bc_range</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_mixFilt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># pareto scaling</span></span>
<span><span class="va">mix_pareto</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_pareto.html">bc_pareto</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_mixFilt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># power scaling</span></span>
<span><span class="va">mix_power</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_power.html">bc_power</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_mixFilt</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># INTERNAL STANDARDS/NEGATIVE CONTROLS</span></span>
<span><span class="co"># RUV-random</span></span>
<span><span class="va">mix_ruv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_ruvRandom.html">bc_ruvRandom</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_mixFilt</span>, nc_cname <span class="op">=</span> <span class="st">"tag"</span>,nc_val <span class="op">=</span> <span class="st">"IS"</span>, k <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># NOMIS</span></span>
<span><span class="va">mix_nomis</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_nomis.html">bc_nomis</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_mixFilt</span>, is_cname <span class="op">=</span> <span class="st">"tag"</span>, is_val <span class="op">=</span> <span class="st">"IS"</span>, num_pc <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># OTHER METHODS</span></span>
<span><span class="va">mix_combat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_combat.html">bc_combat</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_mixFilt</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="mix-data-visualization">Mix: Data Visualization<a class="anchor" aria-label="anchor" href="#mix-data-visualization"></a>
</h3>
<p>Similar to the previous data set, we can compare the PCA plots
between the unadjusted and adjusted data sets.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pmart_mix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">pmart_mixFilt</span>,main_effects <span class="op">=</span> <span class="st">"BatchNum"</span><span class="op">)</span></span>
<span><span class="va">mix_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">mix_range</span>,main_effects <span class="op">=</span> <span class="st">"BatchNum"</span><span class="op">)</span></span>
<span><span class="va">mix_power</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">mix_power</span>,main_effects <span class="op">=</span> <span class="st">"BatchNum"</span><span class="op">)</span></span>
<span><span class="va">mix_pareto</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">mix_pareto</span>,main_effects <span class="op">=</span> <span class="st">"BatchNum"</span><span class="op">)</span></span>
<span><span class="va">mix_combat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">mix_combat</span>,main_effects <span class="op">=</span> <span class="st">"BatchNum"</span><span class="op">)</span></span>
<span><span class="va">mix_ruv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">mix_ruv</span>,main_effects <span class="op">=</span> <span class="st">"BatchNum"</span><span class="op">)</span></span>
<span><span class="va">mix_nomis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/group_designation.html" class="external-link">group_designation</a></span><span class="op">(</span><span class="va">mix_nomis</span>,main_effects <span class="op">=</span> <span class="st">"BatchNum"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pmart_mix</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mix: Unadjusted"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mix_ruv</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mix: RUV-random"</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mix_nomis</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mix: NOMIS"</span><span class="op">)</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mix_combat</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mix: ComBat"</span><span class="op">)</span></span>
<span><span class="va">p5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mix_range</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mix: Range"</span><span class="op">)</span></span>
<span><span class="va">p6</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mix_power</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mix: Power"</span><span class="op">)</span></span>
<span><span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pmartR/man/dim_reduction.html" class="external-link">dim_reduction</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mix_pareto</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_colour_discrete</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Batch"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mix: Pareto"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span>;<span class="va">p2</span>;<span class="va">p3</span>;<span class="va">p4</span>;<span class="va">p5</span>;<span class="va">p6</span>;<span class="va">p7</span></span></code></pre></div>
<p><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-8-1.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-8-2.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-8-3.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-8-4.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-8-5.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-8-6.png" width="33%"><img src="malbacR_vignette_condensed_filtered_files/figure-html/unnamed-chunk-8-7.png" width="33%"></p>
</div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-scaling_paper" class="csl-entry">
Berg, Robert A. van der, Huub CJ Hoefsloot, Johan A. Westerhuis, Age K.
Smilde, and Mariët J. van der Werf. 2006. <span>“Centering, Scaling, and
Transformations: Improving the Biological Information Content of
Metabolomics Data.”</span> <em>BMC Genomics</em> 1: 142. <a href="https://doi.org/10.1186/1471-2164-7-142" class="external-link">https://doi.org/10.1186/1471-2164-7-142</a>.
</div>
<div id="ref-ruvrandom_paper" class="csl-entry">
De Livera, Alysha M., Marko Sysi-Aho, Jacob Laurent, Johann A.
Gagnon-Bartsch, Sandra Castillo, Julie A. Simpson, and Terence P. Speed.
2015. <span>“Statistical Methods for Handling Unwanted Variation in
Metabolomics Data.”</span> <em>Analytical Chemistry</em> 87 (7):
3606–15. <a href="https://doi.org/10.1021/ac502439y" class="external-link">https://doi.org/10.1021/ac502439y</a>.
</div>
<div id="ref-pmartr2" class="csl-entry">
Degnan, David J, Kelly G Stratton, Rachel Richardson, Daniel Claborne,
Evan A Martin, Nathan A Johnson, Damon Leach, Bobbie-Jo M
Webb-Robertson, and Lisa M Bramer. 2023. <span>“pmartR 2.0: A Quality
Control, Visualization, and Statistics Pipeline for Multiple Omics
Datatypes.”</span> <em>Journal of Proteome Research</em> 22 (2): 570–76.
<a href="https://doi.org/10.1021/acs.jproteome.2c00610" class="external-link">https://doi.org/10.1021/acs.jproteome.2c00610</a>.
</div>
<div id="ref-waveica_package" class="csl-entry">
Deng, Kui, and Kang Li. 2022. <em>WaveICA2.0: WaveICA 2.0</em>.
</div>
<div id="ref-waveica_paper" class="csl-entry">
Deng, Kui, Fan Zhang, Qilong Tan, Yue Huang, Zhiwei Rong, Zheng-Jiang
Zhu, Kang Li, and Zhenzi Li. 2019. <span>“WaveICA: A Novel Algorithm to
Remove Batch Effects for Large-Scale Untargeted Metabolomics Data Based
on Wavelet Analysis.”</span> <em>Analytica Chimica Acta</em> 1061:
60–69. <a href="https://doi.org/10.1016/j.aca.2019.02.010" class="external-link">https://doi.org/10.1016/j.aca.2019.02.010</a>.
</div>
<div id="ref-qcrlsc_paper" class="csl-entry">
Dunn, Warwick B., David Broadhurst, Paul Begley, Eva Zelena, Sue
Francis-McIntyrre, Nadine Anderson, Marie Brown, et al. 2011.
<span>“Procedures for Large-Scale Metabolic Profiling of Serum and
Plasma Using Gas Chromatography and Liquid Chromatography Coupled to
Mass Spectrometry.”</span> <em>Nature Protocols</em> 6 (7): 1060–83. <a href="https://doi.org/10.1038/nprot.2011.335" class="external-link">https://doi.org/10.1038/nprot.2011.335</a>.
</div>
<div id="ref-serrf_paper" class="csl-entry">
Fan, Sili, Tobias Kind, Tomas Cajka, Stanley L. Hazen, H. W. Wilson
Tang, Rima Kaddurah-Daouk, Marguerite R. Irvin, Donna K. Arnett, Dinesh
K. Barupal, and Oliver Fiehn. 2019. <span>“Systematic Error Removal
Using Random Forest for Normalizing Large-Scale Untargeted Lipidomics
Data.”</span> <em>Analytical Chemistry</em> 91 (5): 3590–96. <a href="https://doi.org/10.1021/acs.analchem.8b05592" class="external-link">https://doi.org/10.1021/acs.analchem.8b05592</a>.
</div>
<div id="ref-tiger_paper" class="csl-entry">
Han, Siyu, Jialing Huang, Francesco Foppiano, Cornelia Prehn, Jerzy
Adamski, Karsten Suhre, Ying Li, et al. 2022. <span>“TIGER: Technical
Variation Elimination for Metabolomics Data Using Ensemble Learning
Architecture.”</span> <em>Briefings in Bioinformatics</em> 23 (2). <a href="https://doi.org/10.1093/bib/bbab535" class="external-link">https://doi.org/10.1093/bib/bbab535</a>.
</div>
<div id="ref-eigenMS_paper" class="csl-entry">
Karpievitch, Yuliya V., Sonja B. Nikolic, Richard Wilson, James E.
Sharman, and Lindsay M. Edwards. 2014. <span>“Metabolomics Data
Normalization with EigenMS.”</span> <em>PLOS ONE</em> 9 (12). <a href="https://doi.org/10.1371/journal.pone.0116221" class="external-link">https://doi.org/10.1371/journal.pone.0116221</a>.
</div>
<div id="ref-combat_paper" class="csl-entry">
Müller, Christian, Arne Schillert, Caroline Röthmeier, David-Alexandre
Trégouët, Caroline Proust, Harald Binder, Norbert Pfeiffer, et al. 2016.
<span>“Removing Batch Effects from Longitudinal Gene Expression -
Quantile Normalization Plus ComBat as Best Approach for Microarray
Transcriptome Data.”</span> <em>PLOS ONE</em> 11 (6). <a href="https://doi.org/10.1371/journal.pone.0156594" class="external-link">https://doi.org/10.1371/journal.pone.0156594</a>.
</div>
<div id="ref-crmn_package" class="csl-entry">
Redestig, Henning, Atsushi Fukushima, Hans Stenlund, Thomas Moritz,
Masanori Arita, Kazuki Saito, and Miyako Kusano. 2009.
<span>“Compensation for Systematic Cross-Contribution Improves
Normalization of Mass Spectrometry Based Metabolomics Data.”</span>
<em>Analytical Chemistry</em> 81: 7974–80.
</div>
<div id="ref-pmartr" class="csl-entry">
Stratton, Kelly G, Bobbie-Jo M Webb-Robertson, Lee Ann McCue, Daniel
Claborne, Bryan Stanfill, Iobani Godinez, Thomas Johansen, et al. 2019.
<span>“pmartR: Quality Control and Statistics for Mass
Spectrometry-Based Biological Data.”</span> <em>Journal of Proteome
Research</em>.
</div>
<div id="ref-nomis_paper" class="csl-entry">
Sysi-Aho, Marko, Mikko Katajamaa, Laxman Yetukuri, and Matej Orešič.
2007. <span>“Normalization Method for Metabolomics Data Using Optimal
Selection of Multiple Internal Standards.”</span> <em>BMC
Bioinformatics</em> 8: 93. <a href="https://doi.org/10.1186/1471-2105-8-93" class="external-link">https://doi.org/10.1186/1471-2105-8-93</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Damon Leach, Kelly Stratton, Lisa Bramer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
